src/exchange.c

```diff
@@ -231,6 +231,27 @@ enum wps_result do_wps_exchange()
 			set_timeout_is_nack(0);
 
 			ret_val = KEY_REJECTED;
+
+			/* Check the reason code for the received NACK message */
+			if (!get_pin_string_mode()) {
```

Comment:
i'd think we could just unconditionally check for get_nack_reason() == MESSAGE_TIMEOUT here and set ret_val to UNKNOWN_ERROR (assuming that means "try again")